### PR TITLE
Move singleton/lazy initialization of sql connection inside the MySQLDriver

### PR DESCRIPTION
**Pull request by seanim0920**
## Problem
Tests fail sometimes on staging.

## Proposed Solution
This could be due to the sql connection becoming invalid when we recreate the database between tests, or due to the sql connection timing out. I'll add a function in the mysql executable to re-connect to the sql db if the current connection is invalid.

## Implementation Steps
Modify the constructor to create a singleton function that either returns the current db connection or a new db connection if the current one is invalid. To check if the current connection is invalid, mysql provides us a .ping() method.

## Results
Staging tests pass consistently, and this also fixes an open handle error in jest

## Other Changes
Add sleep before running staging tests to allow AWS deployment time to settle
Move recreateDatabase out of dbconnection file
Make MySQLExecutableDriver type NOT include the .execute or .query methods

https://trello.com/c/H29w2JOa/695-back-end-fix-flaky-staging-tests

### CODE REVIEW

**Comment by tifapp** [comment]
Linked Ticket: https://trello.com/c/H29w2JOa

**Comment by seanim0920** [comment]
[Back-End: Fix flaky staging tests](https://trello.com/c/H29w2JOa/695-back-end-fix-flaky-staging-tests)

File: TiFBackendUtils/MySQLDriver/MySQLDriver.ts [hunk 1]
```diff
@@ -34,16 +34,39 @@ const castTypes = (rows: RowDataPacket[], fields: FieldPacket[]): RowDataPacket[
   })
 }
 
-export class MySQLExecutableDriver {
-  private conn: Promise<mysql.Connection>
+export class MySQLDriver {
+  private useConnection: () => Promise<mysql.Connection>
+  private isConnectionClosed: boolean = false
+  
+  constructor () {
```
Comment (seanim0920) [inline TiFBackendUtils/MySQLDriver/MySQLDriver.ts:41]: extract to a separate class

File: TiFBackendUtils/SQLExecutable/tests/MySQLDriver.test.ts [hunk 2]
```diff
@@ -1,34 +1,32 @@
...
 
 export type ExecuteResult = {
   id: string
   rowsAffected: number
 }
 
-describe("MySQLExecutableDriver", () => {
-  let mySQLDriverTest: MySQLExecutableDriver
+describe("MySQLDriver", () => {
+  let mySQLDriverTest: MySQLDriver
```
Comment (seanim0920) [inline TiFBackendUtils/SQLExecutable/tests/MySQLDriver.test.ts:9]: try using public functions to assert connection to db

**Review by suryatiru — DISMISSED** [review]

**Pull request merged**